src/MusicStore/Program.cs

```diff
@@ -59,33 +73,49 @@ public static void Main(string[] args)
                 
                 var minRequestTime = long.MaxValue;
                 var maxRequestTime = long.MinValue;
-                var averageRequestTime = 0.0;
+                int N = 1001;
+                long[] responseTimes = new long[N];
 
-                Console.WriteLine("Running 100 requests");
-                for (var i = 1; i <= 100; i++)
+                Console.WriteLine($"Running {N} requests");
+                JitBenchEventSource.Log.BeginRequests(N);
+                for (var i = 0; i < N; i++)
                 {
                     requestTime.Restart();
                     response = client.GetAsync("http://localhost:5000").Result;
                     requestTime.Stop();
 
-                    var requestTimeElapsed = requestTime.ElapsedMilliseconds;
-                    if (requestTimeElapsed < minRequestTime)
+                    long interval = highRes ? requestTime.ElapsedTicks : requestTime.ElapsedMilliseconds;
+                    responseTimes[i] = interval;
+
+                    if (interval < minRequestTime)
                     {
-                        minRequestTime = requestTimeElapsed;
+                        minRequestTime = interval;
                     }
-
-                    if (requestTimeElapsed > maxRequestTime)
+                    if (interval > maxRequestTime)
                     {
-                        maxRequestTime = requestTimeElapsed;
+                        maxRequestTime = interval;
                     }
-
-                    // Rolling average of request times
-                    averageRequestTime = (averageRequestTime * ((i - 1.0) / i)) + (requestTimeElapsed * (1.0 / i));
                 }
+                JitBenchEventSource.Log.EndRequests(N);
 
-                Console.WriteLine("Steadystate min response time: {0}ms", minRequestTime);
-                Console.WriteLine("Steadystate max response time: {0}ms", maxRequestTime);
-                Console.WriteLine("Steadystate average response time: {0}ms", (int)averageRequestTime);
+                if (highRes)
+                {
+                    double averageResponse = 1000 * ((double)responseTimes.Sum() / N / Stopwatch.Frequency);
+                    double medianResponse = 1000 * ((double)responseTimes.OrderBy(t => t).ElementAt(N / 2) / Stopwatch.Frequency);
+                    Console.WriteLine("Steadystate min response time: {0:F2}ms", (1000 * minRequestTime) / Stopwatch.Frequency);
+                    Console.WriteLine("Steadystate max response time: {0:F2}ms", (1000 * maxRequestTime) / Stopwatch.Frequency);
+                    Console.WriteLine("Steadystate average response time: {0:F2}ms", averageResponse);
+                    Console.WriteLine("Steadystate median response time: {0:F2}ms",  medianResponse);
+                }
+                else
+                {
+                    long averageResponse = responseTimes.Sum() / N;
```

Comment:
Do you know what platforms don't have the HighFrequency counter available? If the case effectively never happens on platforms we care about then we don't have to worry about. If it does show up though...

I'd expect the average and median will generally be an integer between 0-3 which is so course its hard to measure any improvement/regression. What I've done in the past give up on measuring individual requests and instead measure batches of 1000 requests. It doesn't provide a good response distribution but it does give an average with enough precision that you can measure changes.

Reply:
I don't know which platforms are lacking high frequency counters. Let me see if I can find out.

The whole point of using the HF counters is to get at some fractional digits. So in the HF case we'll see median response times like 3.193ms.

Reply:
Yeah I definitely like your proposal to use HF when available, I just wasn't sure how much we needed care about the non-HF case.